Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -35,46 +35,53 @@ extension MongoCollection {
      *
      * - Parameters:
-     *   - documents: The `CollectionType` values to insert
-     *   - options: Optional `InsertManyOptions` to use when executing the command
+     *   - values: The `CollectionType` values to insert
+     *   - options: optional `InsertManyOptions` to use while executing the operation
      *
-     * - Returns: The optional result of attempting to performing the insert. If the write concern
-     *            is unacknowledged, nil is returned
+     * - Returns: an `InsertManyResult`, or `nil` if the write concern is unacknowledged.
+     *
+     * - Throws:
+     *   - `MongoError.invalidArgument` if `values` is empty
+     *   - `MongoError.insertManyError` if any error occurs while performing the writes
      */
     @discardableResult
     public func insertMany(_ values: [CollectionType], options: InsertManyOptions? = nil) throws -> InsertManyResult? {
-        let encoder = BsonEncoder()
+        if values.isEmpty {
+            throw MongoError.invalidArgument(message: "values cannot be empty")
+        }
 
+        let encoder = BsonEncoder()
+        let opts = try encoder.encode(options)
         let documents = try values.map { try encoder.encode($0) }
-        for doc in documents where !doc.keys.contains("_id") {
-            try ObjectId().encode(to: doc.storage, forKey: "_id")
+        var insertedIds: [Int: BsonValue?] = [:]
+
+        try documents.enumerated().forEach { (index, document) in
+            if !document.keys.contains("_id") {
+                try ObjectId().encode(to: document.storage, forKey: "_id")
+            }
+            insertedIds[index] = document["_id"]
         }
-        var docPointers = documents.map { UnsafePointer($0.data) }
 
-        let opts = try encoder.encode(options)
+        var docPointers = documents.map { UnsafePointer($0.data) }
         let reply = Document()
         var error = bson_error_t()
-        if !mongoc_collection_insert_many(
-            self._collection, &docPointers, documents.count, opts?.data, reply.data, &error) {
-            // TODO SWIFT-139: include writeErrors and writeConcernErrors from reply in the error
-            throw MongoError.commandError(message: toErrorString(error))
-        }
 
-        guard isAcknowledged(options?.writeConcern) else {
-            return nil
-        }
+        let success = mongoc_collection_insert_many(self._collection, &docPointers, values.count, opts?.data, reply.data, &error)
+        let result = try InsertManyResult(reply: reply, insertedIds: insertedIds)
+        let isAcknowledged = self.isAcknowledged(options?.writeConcern)
 
-        // if we succeeded, then we should store + return every `_id`
-        var insertedIds = [Int: BsonValue?]()
-        for (index, doc) in documents.enumerated() {
-            insertedIds[index] = doc["_id"]
+        guard success else {
+            throw MongoError.insertManyError(code: error.code, message: toErrorString(error),
```

Comment:
nit: message on new line

Reply:
I thought this was kosher because we're just abiding a 120-character wrap and not necessarily placing each argument on its own line. We were doing the same in https://github.com/mongodb/mongo-swift-driver/blob/185829de1358094b25838781c126db000cad8a7c/Sources/MongoSwift/MongoCollection%2BBulkWrite.swift#L281.

Should I change both?